src/loader.c

```diff
@@ -14,6 +15,335 @@ uint8_t mcb_alloc_st = 0;
+                break;
+            } 
+            break;
+        case FCB_PARSE_FCB1:
+            switch (cmd_line[i]) {
+            case '.':
+                offset = fcb1 + 9;
+                state = FCB_PARSE_FCB1_EXT;
+                break;
+            case '*':
+                while (offset - fcb1 - 1 < 8) {
+                    *offset = '?';
+                    offset++;
+                }
+                break;
+#if defined(FCB_DOS11)
+            case '+':
+            case ';':
+                offset = fcb2 + 1;
+                state = FCB_PARSE_SEP_PLUS;
+                break;
+            case ':':
+                offset = fcb2 + 1;
+                state = FCB_PARSE_FCB2;
+                break;
+#endif
+            default:
+                if (valid_fcb_sep(cmd_line[i])) {
+                    offset = fcb2 + 1;
+                    state = FCB_PARSE_SEP;
+                    break;
+                }
+                if (!valid_fcb_char(cmd_line[i])) {
+                    offset = fcb2 + 1;
+#if defined(FCB_DOS11) 
+                    state = FCB_PARSE_EXIT;
+#else
+                    offset = fcb2 + 1;
+                    state = FCB_PARSE_SEP_PURGE;
+#endif
+                    break;
+                }
+                if (offset - fcb1 - 1 < 8) {
+                    *offset = toupper(cmd_line[i]);
+                    offset++;
+                }
+                break;
+            }
+            break;
+        case FCB_PARSE_FCB1_EXT:
+            switch (cmd_line[i]) {
+            case '.':
+#if defined(FCB_DOS11)
+                offset = fcb2 + 9;
+                state = FCB_PARSE_FCB2_EXT;
+#else
+                offset = fcb2 + 1;
+                state = FCB_PARSE_SEP_PURGE;
+#endif
+                break;
+            case '*':
+                while (offset - fcb1 - 9 < 3) {
+                    *offset = '?';
+                    offset++;
+                }
+                break;
+#if defined(FCB_DOS11)
+            case ';':
+            case '+':
+                offset = fcb2 + 1;
+                state = FCB_PARSE_SEP_PLUS;
+                break;
+            case ':':
+                offset = fcb2 + 1;
+                state = FCB_PARSE_FCB2;
+                break;
+#endif
+            default:
+                if (valid_fcb_sep(cmd_line[i])) {
+                    offset = fcb2 + 1;
+                    state = FCB_PARSE_SEP;
+                    break;
+                }
+                if (!valid_fcb_char(cmd_line[i])) {
+#if defined(FCB_DOS11)
+                    state = FCB_PARSE_EXIT;
+#else
+                    offset = fcb2 + 1;
+                    state = FCB_PARSE_SEP_PURGE;
+#endif
+                    break;
+                }
+                if (offset - fcb1 - 9 < 3) {
+                    *offset = toupper(cmd_line[i]);
+                    offset++;
+                }
+                break;
+            }
+            break;
+#if !defined(FCB_DOS11)
```

Comment:
You don't need the `#if`, as the FCB_PARSE_SEP_PURGE state can't be reached in DOS 2 or 3 modes.

Reply:
Done